### PR TITLE
CMake script rework

### PR DESCRIPTION
**Pull request by rokups**
I took a liberty to clean up cmake build scripts as they were quite a mess :p

You had project options to be defined top-down (using `include_directories`, `add_definitions` and setting `CMAKE_(C|CXX)_FLAGS`) which is terrible for libraries. I reworked scripts so that build options and definitions are bottom-up (using `target_include_directories`, `target_compile_definitions` and `target_compile_options`). Meaning any project that consumes `easy_profiler` gets it's exported definitions. For example `profiler_gui` on it's own defined profiler version definitions, this is no longer needed as any project that links to library gets these auto-magically.

Cmake version requirement got slightly increased from 2.8 to 2.8.12.1 (which is shipped with ubuntu 14.04 LTS), however `profiler_gui` no longer needs 3+ version.

I removed `LIB_NAME` variable from `easy_profiler_core`. It seemed to serve a purpose of customizing lib name. I am not sure why it would be required, if it is for changing output lib name - it can be done by setting `easy_profiler` target properties and this should be preferred way.

Overall code got leaner as i could get rid of lots of boilerplate code by exploiting that target property inheritance i mentioned and by using a macro for adding some definitions that otherwise were handled with endless `if/else` cases.

To sum it up: what we get is, in addition to classic install, ability to consume library in other cmake projects by simply using `add_subdirectory(easy_profiler)` and `target_link_libraries(target easy_profiler)`. 

Edit:
Forgot to mention that i removed `version.info` file and version is set in main cmake script. It is way simpler that way and i could not see any reason why version should be tucked away in some file. Hope that was a right call.

### CODE REVIEW

**Comment by rokups** [comment]
Hope it all works, if there are any problems let me know.
